syntax = "proto3";

message User {
  string first_name = 1;
  string last_name = 2;
  string email = 3;
  string user_id = 4;
}

message UserList {
  repeated User users = 1;
}

message Empty {

}

message RequestById {
  string user_id = 1;
}

service UserService {
  rpc GetUserList(Empty) returns (UserList) {}
  rpc GetUserById(RequestById) returns (User) {}
  rpc UpdateUser(RequestById) returns (User) {}
}